service/matching/matcher_test.go

```diff
@@ -95,8 +96,8 @@ func (t *MatcherTestSuite) SetupTest() {
 	t.Assert().NoError(err)
 	t.matcher = newTaskMatcher(tlCfg, t.fwdr, metrics.NoopMetricsHandler)
 
-	rootTaskqueueCfg := newTaskQueueConfig(prtn.TaskQueue(), cfg, "test-namespace")
-	t.rootMatcher = newTaskMatcher(rootTaskqueueCfg, nil, metrics.NoopMetricsHandler)
+	t.rootcfg = newTaskQueueConfig(prtn.TaskQueue(), cfg, "test-namespace")
```

Comment:
it doesn't feel right to have two config objects for the same TQ. config is per TQ, not per partition. I think the code was slightly wrong before. I'd replace the lines 86 onwards with the following:
```
	t.cfg = newTaskQueueConfig(prtn.TaskQueue(), cfg, "test-namespace")
	t.cfg.forwarderConfig = forwarderConfig{
		ForwarderMaxOutstandingPolls: func() int { return 1 },
		ForwarderMaxOutstandingTasks: func() int { return 1 },
		ForwarderMaxRatePerSecond:    func() int { return 2 },
		ForwarderMaxChildrenPerNode:  func() int { return 20 },
	}
	t.fwdr, err = newForwarder(&t.cfg.forwarderConfig, t.queue, t.client)
	t.Assert().NoError(err)
	t.matcher = newTaskMatcher(t.cfg, t.fwdr, metrics.NoopMetricsHandler)
	t.rootMatcher = newTaskMatcher(t.cfg, nil, metrics.NoopMetricsHandler)
```

Reply:
There's two levels of config object, `cfg` is the "global" one, the return value of `newTaskQueueConfig` is the per-partition one. And it is per-partition even if it's identical for different partitions of the same tq.. in a real system there would be one of those per partition, probably in different processes. They happen to be in the same process here since it's a unit test.

Reply:
OK. fair. then maybe we should rename `t.matcher` -> `t.childMatcher` and `t.cfg` -> `t.childCfg` so test author does not change `t.cfg` when they intend to change `t.rootCfg`.

Reply:
Done. I also changed cfg to config to follow our convention of spelling things out